specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,963 @@
+| `getelementptr inbounds` | Used to create an `i8*` to pass a constant string for the purpose of labeling an output value.    | May be used as part of a call to an output recording function only.                                         |
+
+The adaptive profile extends the base profile with additional instructions such
+as `br` (**Bullet 4**) and the additional opt-in instructions or intrinsics in
+(**Bullet 5**)
+
+See also the section on [data types and values](#data-types-and-values) for more
+information about the creation and usage of LLVM values.
+
+## Run-Time Functions
+
+The following run-time functions must be supported by all backends:
+
+| Function                            | Signature            | Description                                                                                                                                                                                                                                                  |
+| :---------------------------------- | :------------------- | :----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| __quantum__rt__initialize           | `void(i8*)`          | Initializes the execution environment. Sets all qubits to a zero-state if they are not dynamically managed.                                                                                                                                                 |
+| __quantum__rt__tuple_record_output  | `void(i64,i8*)`      | Inserts a marker in the generated output that indicates the start of a tuple and how many tuple elements it has. The second parameter defines a string label for the tuple. Depending on the output schema, the label is included in the output or omitted.  |
+| __quantum__rt__array_record_output  | `void(i64,i8*)`      | Inserts a marker in the generated output that indicates the start of an array and how many array elements it has. The second parameter defines a string label for the array. Depending on the output schema, the label is included in the output or omitted. |
+| __quantum__rt__result_record_output | `void(%Result*,i8*)` | Adds a measurement result to the generated output. The second parameter defines a string label for the result value. Depending on the output schema, the label is included in the output or omitted.                                                         |
+| __quantum__rt__bool_record_output   | `void(i1,i8*)`       | Adds a boolean value to the generated output. The second parameter defines a string label for the result value. Depending on the output schema, the label is included in the output or omitted.                                                              |
+
```

Comment:
We need to update the output recording formats to include any changes in recording varying bit widths for int, float, fixed.